nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,282 @@
+
+    public static final PropertyDescriptor DATABASE_NAME = new PropertyDescriptor.Builder()
+            .name("Database Name")
+            .displayName("Database Name")
+            .description("Azure Data Explorer Database Name for querying")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+    public static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("Table Name")
+            .displayName("Table Name")
+            .description("Azure Data Explorer Table Name for ingesting data")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+    public static final PropertyDescriptor MAPPING_NAME = new PropertyDescriptor
+            .Builder().name("Ingest Mapping name")
+            .displayName("Ingest Mapping name")
+            .description("The name of the mapping responsible for storing the data in the appropriate columns.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor IS_STREAMING_ENABLED = new PropertyDescriptor
+            .Builder().name("Streaming Enabled")
+            .displayName("Streaming Enabled")
+            .description("This property determines whether we want to stream data to ADX.")
+            .required(false)
+            .allowableValues(Boolean.TRUE.toString(), Boolean.FALSE.toString())
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .defaultValue(Boolean.FALSE.toString())
+            .build();
+    public static final PropertyDescriptor POLL_ON_INGESTION_STATUS = new PropertyDescriptor
+            .Builder().name("Poll for Ingestion Status")
+            .displayName("Poll for Ingestion Status")
+            .description("Determines whether to poll on ingestion status after an ingestion to Azure Data Explorer is completed")
+            .required(false)
+            .allowableValues(Boolean.TRUE.toString(), Boolean.FALSE.toString())
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .defaultValue(Boolean.FALSE.toString())
+            .build();
+    public static final PropertyDescriptor ADX_SERVICE = new PropertyDescriptor
+            .Builder().name("Kusto Ingest Service")
+            .displayName("Kusto Ingest Service")
+            .description("Azure Data Explorer Kusto Ingest Service")
+            .required(true)
+            .identifiesControllerService(KustoIngestService.class)
+            .build();
+    public static final PropertyDescriptor DATA_FORMAT = new PropertyDescriptor.Builder()
+            .name("Data Format")
```

Comment:
What do you think about supporting Expression Language for the Data Format property? This would allow the property value to be resolved based on a FlowFile attribute, so that one Processor could handle different types of data. That would require call `evaluateAttributeExpressions(flowFile)` in the `onTrigger()` method.

Reply:
yeah makes sense.